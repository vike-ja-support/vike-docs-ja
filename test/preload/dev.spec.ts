import { beforeAll } from 'vitest'
import { createServer } from 'vite'
import { testRun } from './testRun'

beforeAll(async () => {
  await devApp()
}, 10 * 1000)

testRun(true)

async function devApp() {
  await createServer({
    root: __dirname,
    server: {
      middlewareMode: true,
      /* Doesn't seem to work
      hmr: false
      */
      hmr: {
        // Avoid:
        // ```
        // Error: listen EADDRINUSE: address already in use :::24678
        // ```
        port: 11323
      }
    }
  })
}
