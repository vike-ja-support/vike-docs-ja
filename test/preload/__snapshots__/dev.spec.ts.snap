// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`preload tags > Preload Default 1`] = `
[
  {
    "assetType": "script",
    "earlyHintLink": "</@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js>; rel=modulepreload; as=script",
    "isEntry": true,
    "mediaType": "text/javascript",
    "src": "/@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js",
  },
  {
    "assetType": "style",
    "earlyHintLink": "</renderer/PageLayout.css?direct>; rel=preload; as=style",
    "isEntry": true,
    "mediaType": "text/css",
    "src": "/renderer/PageLayout.css?direct",
  },
]
`;

exports[`preload tags > Preload Default 2`] = `
"<!DOCTYPE html>
    <html>
      <head>
        <link rel="stylesheet" type="text/css" href="/renderer/PageLayout.css?direct">
      </head>
      <body>
        <div id="page-view"><div style="display:flex;max-width:900px;margin:auto"><div style="padding:20px;padding-top:20px;flex-shrink:0;display:flex;flex-direction:column;align-items:center;line-height:1.8em"><div style="margin-top:20px;margin-bottom:10px"><a href="/"><img src="/renderer/logo.svg" height="64" width="64"/></a></div><a class="navitem" href="/">Preload Default</a><a class="navitem" href="/preload-disabled">Preload Disabled</a><a class="navitem" href="/preload-images">Preload Images</a></div><div style="padding:20px;padding-bottom:50px;border-left:2px solid #eee;min-height:100vh"><h1>Default</h1><p>This page showcases the default preloading strategy: in production, both the image and the font are preloaded.</p></div></div></div>
        <script id="vike_pageContext" type="application/json">{"pageProps":"!undefined","abortReason":"!undefined","_urlRewrite":null,"_urlRedirect":"!undefined","abortStatusCode":"!undefined","_abortCall":"!undefined","_pageContextInitIsPassedToClient":"!undefined","_pageId":"/pages/index","data":"!undefined"}</script>
        <script type="module" async>
import RefreshRuntime from "/@react-refresh"
RefreshRuntime.injectIntoGlobalHook(window)
window.$RefreshReg$ = () => {}
window.$RefreshSig$ = () => (type) => type
window.__vite_plugin_react_preamble_installed__ = true
import("/@vite/client");
import("/@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js");
</script>
      </body>
    </html>"
`;

exports[`preload tags > Preload Disabled 1`] = `
[
  {
    "assetType": "script",
    "earlyHintLink": "</@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js>; rel=modulepreload; as=script",
    "isEntry": true,
    "mediaType": "text/javascript",
    "src": "/@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js",
  },
  {
    "assetType": "style",
    "earlyHintLink": "</renderer/PageLayout.css?direct>; rel=preload; as=style",
    "isEntry": true,
    "mediaType": "text/css",
    "src": "/renderer/PageLayout.css?direct",
  },
]
`;

exports[`preload tags > Preload Disabled 2`] = `
"<!DOCTYPE html>
    <html>
      <head>
        <link rel="stylesheet" type="text/css" href="/renderer/PageLayout.css?direct">
      </head>
      <body>
        <div id="page-view"><div style="display:flex;max-width:900px;margin:auto"><div style="padding:20px;padding-top:20px;flex-shrink:0;display:flex;flex-direction:column;align-items:center;line-height:1.8em"><div style="margin-top:20px;margin-bottom:10px"><a href="/"><img src="/renderer/logo.svg" height="64" width="64"/></a></div><a class="navitem" href="/">Preload Default</a><a class="navitem" href="/preload-disabled">Preload Disabled</a><a class="navitem" href="/preload-images">Preload Images</a></div><div style="padding:20px;padding-bottom:50px;border-left:2px solid #eee;min-height:100vh"><h1>Preload Disabled</h1><p>This page showcases completely disabled preloading: the image nor the font are preloaded.</p></div></div></div>
        <script id="vike_pageContext" type="application/json">{"pageProps":"!undefined","abortReason":"!undefined","_urlRewrite":null,"_urlRedirect":"!undefined","abortStatusCode":"!undefined","_abortCall":"!undefined","_pageContextInitIsPassedToClient":"!undefined","_pageId":"/pages/preload-disabled","data":"!undefined"}</script>
        <script type="module" async>
import RefreshRuntime from "/@react-refresh"
RefreshRuntime.injectIntoGlobalHook(window)
window.$RefreshReg$ = () => {}
window.$RefreshSig$ = () => (type) => type
window.__vite_plugin_react_preamble_installed__ = true
import("/@vite/client");
import("/@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js");
</script>
      </body>
    </html>"
`;

exports[`preload tags > Preload Eager 1`] = `
[
  {
    "assetType": "script",
    "earlyHintLink": "</@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js>; rel=modulepreload; as=script",
    "isEntry": true,
    "mediaType": "text/javascript",
    "src": "/@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js",
  },
  {
    "assetType": "style",
    "earlyHintLink": "</renderer/PageLayout.css?direct>; rel=preload; as=style",
    "isEntry": true,
    "mediaType": "text/css",
    "src": "/renderer/PageLayout.css?direct",
  },
]
`;

exports[`preload tags > Preload Eager 2`] = `
"<!DOCTYPE html>
    <html>
      <head>
        <link rel="stylesheet" type="text/css" href="/renderer/PageLayout.css?direct">
      </head>
      <body>
        <div id="page-view"><div style="display:flex;max-width:900px;margin:auto"><div style="padding:20px;padding-top:20px;flex-shrink:0;display:flex;flex-direction:column;align-items:center;line-height:1.8em"><div style="margin-top:20px;margin-bottom:10px"><a href="/"><img src="/renderer/logo.svg" height="64" width="64"/></a></div><a class="navitem" href="/">Preload Default</a><a class="navitem" href="/preload-disabled">Preload Disabled</a><a class="navitem" href="/preload-images">Preload Images</a></div><div style="padding:20px;padding-bottom:50px;border-left:2px solid #eee;min-height:100vh"><h1>Eager</h1><p>This page showcases eager preloading (non-JavaScript assets are preloaded ASAP).</p></div></div></div>
        <script id="vike_pageContext" type="application/json">{"pageProps":"!undefined","abortReason":"!undefined","_urlRewrite":null,"_urlRedirect":"!undefined","abortStatusCode":"!undefined","_abortCall":"!undefined","_pageContextInitIsPassedToClient":"!undefined","_pageId":"/pages/preload-eager","data":"!undefined"}</script>
        <script type="module" async>
import RefreshRuntime from "/@react-refresh"
RefreshRuntime.injectIntoGlobalHook(window)
window.$RefreshReg$ = () => {}
window.$RefreshSig$ = () => (type) => type
window.__vite_plugin_react_preamble_installed__ = true
import("/@vite/client");
import("/@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js");
</script>
      </body>
    </html>"
`;

exports[`preload tags > Preload Images 1`] = `
[
  {
    "assetType": "script",
    "earlyHintLink": "</@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js>; rel=modulepreload; as=script",
    "isEntry": true,
    "mediaType": "text/javascript",
    "src": "/@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js",
  },
  {
    "assetType": "style",
    "earlyHintLink": "</renderer/PageLayout.css?direct>; rel=preload; as=style",
    "isEntry": true,
    "mediaType": "text/css",
    "src": "/renderer/PageLayout.css?direct",
  },
]
`;

exports[`preload tags > Preload Images 2`] = `
"<!DOCTYPE html>
    <html>
      <head>
        <link rel="stylesheet" type="text/css" href="/renderer/PageLayout.css?direct">
      </head>
      <body>
        <div id="page-view"><div style="display:flex;max-width:900px;margin:auto"><div style="padding:20px;padding-top:20px;flex-shrink:0;display:flex;flex-direction:column;align-items:center;line-height:1.8em"><div style="margin-top:20px;margin-bottom:10px"><a href="/"><img src="/renderer/logo.svg" height="64" width="64"/></a></div><a class="navitem" href="/">Preload Default</a><a class="navitem" href="/preload-disabled">Preload Disabled</a><a class="navitem" href="/preload-images">Preload Images</a></div><div style="padding:20px;padding-bottom:50px;border-left:2px solid #eee;min-height:100vh"><h1>Preload Images</h1><p>This page showcases a custom strategy of preloading images.</p></div></div></div>
        <script id="vike_pageContext" type="application/json">{"pageProps":"!undefined","abortReason":"!undefined","_urlRewrite":null,"_urlRedirect":"!undefined","abortStatusCode":"!undefined","_abortCall":"!undefined","_pageContextInitIsPassedToClient":"!undefined","_pageId":"/pages/preload-images","data":"!undefined"}</script>
        <script type="module" async>
import RefreshRuntime from "/@react-refresh"
RefreshRuntime.injectIntoGlobalHook(window)
window.$RefreshReg$ = () => {}
window.$RefreshSig$ = () => (type) => type
window.__vite_plugin_react_preamble_installed__ = true
import("/@vite/client");
import("/@fs/$ROOT/vike/dist/esm/client/server-routing-runtime/entry.js");
</script>
      </body>
    </html>"
`;
